**Target Company : SBIN** Tickers given in the brackets.

Comparable Companies:
1. Axis Bank(AXISBANK)
2. HDFC Bank(HDFCBANK)
3. ICICI Bank(ICICIBANK)
4. Kotak Mahindra Bank(KOTAKBANK)

In [ ]:
#Importing the required libraries.
!pip install yfinance
import matplotlib.pyplot as plot
import statsmodels.api as stm
import yfinance as yfin
import pandas as pds
import numpy as npy
import seaborn as sbn
import scipy.stats as stat


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
#Using the stock prices of monthly data of security for last 5 years
sbin_df = yfin.download('SBIN.NS',
                      start='2017-11-01',
                      end='2022-11-01',
                      progress=False,interval ='1mo'
)
sbin_df.head()

Open        High         Low       Close   Adj Close  \
Date                                                                     
2017-11-01  309.399994  346.700012  306.649994  320.350006  312.541870   
2017-12-01  321.500000  323.850006  300.399994  309.899994  302.346558   
2018-01-01  310.600006  332.750000  292.600006  313.250000  305.614899   
2018-02-01  316.899994  317.399994  256.649994  268.000000  261.467804   
2018-03-01  266.299988  270.000000  232.350006  249.899994  243.808975   

               Volume  
Date                   
2017-11-01  523180524  
2017-12-01  308352494  
2018-01-01  467226485  
2018-02-01  506361276  
2018-03-01  491062494

In [ ]:
#Downloading the closing values for the Nifty index
niftybse_df = yfin.download('^BSESN',start='2017-11-01',
                      end='2022-11-01',
                      progress=False,interval ='1mo',
)
niftybse_df.head()

Open          High           Low         Close  \
Date                                                                 
2017-11-01  33344.230469  33865.949219  32683.589844  33149.351562   
2017-12-01  33247.660156  34137.968750  32565.160156  34056.828125   
2018-01-01  34059.988281  36443.980469  33703.371094  35965.019531   
2018-02-01  36048.988281  36256.828125  33482.808594  34184.039062   
2018-03-01  34141.218750  34278.628906  32483.839844  32968.679688   

               Adj Close  Volume  
Date                              
2017-11-01  33149.351562  718100  
2017-12-01  34056.828125  252700  
2018-01-01  35965.019531  382500  
2018-02-01  34184.039062  342600  
2018-03-01  32968.679688  266400

In [ ]:
# Determining the log returns of nifty
niftybse_df['Return_NIFTY'] = npy.log(niftybse_df['Adj Close']/niftybse_df['Adj Close'].shift(1))
niftybse_df.head()
niftybse_df1=niftybse_df.dropna(axis=0)
niftybse_df1.head()


Open          High           Low         Close  \
Date                                                                 
2017-12-01  33247.660156  34137.968750  32565.160156  34056.828125   
2018-01-01  34059.988281  36443.980469  33703.371094  35965.019531   
2018-02-01  36048.988281  36256.828125  33482.808594  34184.039062   
2018-03-01  34141.218750  34278.628906  32483.839844  32968.679688   
2018-04-01  33030.871094  35213.300781  32972.558594  35160.359375   

               Adj Close  Volume  Return_NIFTY  
Date                                            
2017-12-01  34056.828125  252700      0.027007  
2018-01-01  35965.019531  382500      0.054516  
2018-02-01  34184.039062  342600     -0.050788  
2018-03-01  32968.679688  266400     -0.036201  
2018-04-01  35160.359375  227700      0.064361

In [ ]:
# Calculating the log returns for sbi bank
sbin_df['Return_Sbi']=npy.log(sbin_df['Adj Close']/sbin_df['Adj Close'].shift(1))
sbin_df.head()
sbin_df1 = sbin_df.dropna(axis=0)
sbin_df1.head()

Open        High         Low       Close   Adj Close  \
Date                                                                     
2017-12-01  321.500000  323.850006  300.399994  309.899994  302.346558   
2018-01-01  310.600006  332.750000  292.600006  313.250000  305.614899   
2018-02-01  316.899994  317.399994  256.649994  268.000000  261.467804   
2018-03-01  266.299988  270.000000  232.350006  249.899994  243.808975   
2018-04-01  251.800003  265.000000  232.500000  246.399994  240.394287   

               Volume  Return_Sbi  
Date                               
2017-12-01  308352494   -0.033165  
2018-01-01  467226485    0.010752  
2018-02-01  506361276   -0.156015  
2018-03-01  491062494   -0.069926  
2018-04-01  387131760   -0.014105

In [ ]:
# Placing the returns of SBI and NIFTY into columns
sbin_df1_returns = sbin_df1['Return_Sbi']
NIFTY_df1_returns = niftybse_df1['Return_NIFTY']
# Merging the columns of returns
returns= pds.merge(sbin_df1_returns,NIFTY_df1_returns,how='inner',on='Date')
returns.head()
returns.describe()

Return_Sbi  Return_NIFTY
count   59.000000     59.000000
mean     0.010297      0.010266
std      0.116283      0.057637
min     -0.431291     -0.262058
25%     -0.042751     -0.020434
50%      0.005556      0.009231
75%      0.084035      0.045031
max      0.324270      0.134699

- *Annualised Returns_SBI = 12.35% ( We are taking APR across all returns, so Annualised Returns are calculated as 12 * mean Returns from above table)*
- *Annualised Returns_NIFTY = 12.32%*

Market return(Rm)=12.32%


In [ ]:
# Declaring Regression Variables
x=returns['Return_NIFTY']
y=returns['Return_Sbi']
x_sm=stm.add_constant(x)

In [ ]:
#Regression
model = stm.OLS(y,x_sm)
#OLS estimation or Ordinary Least square Minimizes the sum of squared errors
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             Return_Sbi   R-squared:                       0.495
Model:                            OLS   Adj. R-squared:                  0.486
Method:                 Least Squares   F-statistic:                     55.93
Date:                Mon, 24 Apr 2023   Prob (F-statistic):           5.06e-10
Time:                        05:10:49   Log-Likelihood:                 63.910
No. Observations:                  59   AIC:                            -123.8
Df Residuals:                      57   BIC:                            -119.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           -0.0043      0.011     -0.388      0.699      -0.026       0.018
Return_NIFTY     1.4199      0.190      7.479      0.000       1.040       1.800
==============================================================================
Omnibus:                        2.150   Durbin-Watson:                   2.098
Prob(Omnibus):                  0.341   Jarque-Bera (JB):                1.415
Skew:                           0.134   Prob(JB):                        0.493
Kurtosis:                       3.709   Cond. No.                         17.5
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## Beta levered according to top down approach for SBI stock = 1.4199 and significant at 5%

- Calculating the Cost of equity according to top down approach using CAPM the model
- Market risk(Rm) = 0.1232
- Risk-free rate(Rf) = 0.0475=4.57%
- Market Risk Premium = 0.1232-0.0475 = 0.0757=7.57%
- *Cost_of_eq = 0.0475 + 1.4199(0.0757)= 0.15498 i.e 15.49%*


# Bottom up approach by unlevering and then relevering the beta

In [ ]:
#axis bank
#Using the stock prices of monthly securities for last 5 years
axis_df = yfin.download('AXISBANK.NS',
                      start='2017-11-01',
                      end='2022-11-01',
                      progress=False,interval ='1mo'
)
#Calculating the log returns for axis bank
axis_df['Return_AXIS'] = npy.log(axis_df['Adj Close']/axis_df['Adj Close'].shift(1))
axis_df1 = axis_df.dropna(axis=0) # Removing the null values
axis_df1.head()
print(axis_df['Return_AXIS'])
axis_df1 = axis_df.dropna(axis=0)
axis_df1_returns = axis_df1['Return_AXIS']
axis_df1_returns.head()
returns = pds.merge(NIFTY_df1_returns,axis_df1_returns,how='inner',on='Date')
returns.head()
returns.describe()
x = returns['Return_NIFTY']
y = returns['Return_AXIS']
x_sm = stm.add_constant(x)
#Regression --- Market model
model = stm.OLS(y,x_sm)
results = model.fit()
#Regression Result(summary):
results.summary()


Date
2017-11-01         NaN
2017-12-01    0.051951
2018-01-01    0.051240
2018-02-01   -0.115690
2018-03-01   -0.035125
2018-04-01    0.013232
2018-05-01    0.053813
2018-06-01   -0.066458
2018-07-01    0.074667
2018-08-01    0.165173
2018-09-01   -0.057045
2018-10-01   -0.051615
2018-11-01    0.071794
2018-12-01   -0.009393
2019-01-01    0.153436
2019-02-01   -0.018363
2019-03-01    0.091131
2019-04-01   -0.013471
2019-05-01    0.052642
2019-06-01    0.000309
2019-07-01   -0.181864
2019-08-01   -0.014006
2019-09-01    0.031287
2019-10-01    0.072287
2019-11-01    0.003660
2019-12-01    0.020159
2020-01-01   -0.033440
2020-02-01   -0.044869
2020-03-01   -0.609679
2020-04-01    0.160313
2020-05-01   -0.144736
2020-06-01    0.054839
2020-07-01    0.059662
2020-08-01    0.140472
2020-09-01   -0.156822
2020-10-01    0.148229
2020-11-01    0.200098
2020-12-01    0.030852
2021-01-01    0.066179
2021-02-01    0.089271
2021-03-01   -0.038465
2021-04-01    0.024712
2021-05-01    0.048863
2021-0

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            Return_AXIS   R-squared:                       0.708
Model:                            OLS   Adj. R-squared:                  0.703
Method:                 Least Squares   F-statistic:                     138.3
Date:                Sun, 20 Nov 2022   Prob (F-statistic):           7.15e-17
Time:                        17:35:24   Log-Likelihood:                 78.757
No. Observations:                  59   AIC:                            -153.5
Df Residuals:                      57   BIC:                            -149.4
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           -0.0089      0.009     -1.033      0.306      -0.026       0.008
Return_NIFTY     1.7356      0.148     11.758      0.000       1.440       2.031
==============================================================================
Omnibus:                        2.337   Durbin-Watson:                   2.123
Prob(Omnibus):                  0.311   Jarque-Bera (JB):                1.751
Skew:                           0.417   Prob(JB):                        0.417
Kurtosis:                       3.134   Cond. No.                         17.5
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

- *Beta levered for AXIS Bank is 1.7356*
- *D/E ratio for Axis bank is 8.64*
- *Unlevered Beta for AXis Bank is =  1.7356/(1+.65X(Debt/MVE)) = 1.7356/(1+.65X(8.64))=0.2623*


In [ ]:
#hdfc bank
#Downloading the stock prices for the security monthly data last 5 years
hdfc_df = yfin.download('HDFCBANK.NS',
                      start='2017-11-01',
                      end='2022-11-01',
                      progress=False,interval ='1mo'
)
#Calculating the log returns for hdfc bank
hdfc_df['Return_HDFC'] = npy.log(hdfc_df['Adj Close']/hdfc_df['Adj Close'].shift(1))
hdfc_df1 = hdfc_df.dropna(axis=0) # removing the null values
hdfc_df1.head()
print(hdfc_df['Return_HDFC'])
hdfc_df1 = hdfc_df.dropna(axis=0)
hdfc_df1_returns = hdfc_df1['Return_HDFC']
hdfc_df1_returns.head()
returns = pds.merge(NIFTY_df1_returns,hdfc_df1_returns,how='inner',on='Date')
returns.head()
returns.describe()
x = returns['Return_NIFTY']
y = returns['Return_HDFC']
x_sm = stm.add_constant(x)
#Regression --- Market model
model = stm.OLS(y,x_sm)
results = model.fit()
#Regression result(summary)
results.summary()


Date
2017-11-01         NaN
2017-12-01    0.010037
2018-01-01    0.068772
2018-02-01   -0.062490
2018-03-01    0.001008
2018-04-01    0.030391
2018-05-01    0.095647
2018-06-01   -0.008229
2018-07-01    0.033142
2018-08-01   -0.055807
2018-09-01   -0.027121
2018-10-01   -0.048149
2018-11-01    0.107375
2018-12-01   -0.003176
2019-01-01   -0.019874
2019-02-01   -0.001154
2019-03-01    0.109904
2019-04-01   -0.000625
2019-05-01    0.045508
2019-06-01    0.007558
2019-07-01   -0.075674
2019-08-01   -0.010581
2019-09-01    0.099227
2019-10-01    0.002360
2019-11-01    0.035608
2019-12-01   -0.002238
2020-01-01   -0.036668
2020-02-01   -0.040481
2020-03-01   -0.312137
2020-04-01    0.150414
2020-05-01   -0.051356
2020-06-01    0.113331
2020-07-01   -0.031499
2020-08-01    0.077343
2020-09-01   -0.033952
2020-10-01    0.092854
2020-11-01    0.196715
2020-12-01   -0.003163
2021-01-01   -0.032407
2021-02-01    0.098476
2021-03-01   -0.026917
2021-04-01   -0.056003
2021-05-01    0.070757
2021-0

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            Return_HDFC   R-squared:                       0.738
Model:                            OLS   Adj. R-squared:                  0.734
Method:                 Least Squares   F-statistic:                     160.9
Date:                Sun, 20 Nov 2022   Prob (F-statistic):           3.07e-18
Time:                        17:36:15   Log-Likelihood:                 110.57
No. Observations:                  59   AIC:                            -217.1
Df Residuals:                      57   BIC:                            -213.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           -0.0026      0.005     -0.514      0.609      -0.013       0.007
Return_NIFTY     1.0919      0.086     12.685      0.000       0.920       1.264
==============================================================================
Omnibus:                        2.876   Durbin-Watson:                   2.615
Prob(Omnibus):                  0.237   Jarque-Bera (JB):                2.279
Skew:                          -0.161   Prob(JB):                        0.320
Kurtosis:                       3.907   Cond. No.                         17.5
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

- *Beta levered for HDFC Bank is 1.0919*
- *D/E ratio for HDFC bank is 7.22*
- *Unlevered Beta for HDFC Bank is =  1.0919/(1+.65X(Debt/MVE)) = 1.0919/(1+.65X(7.22))= 0.1917*


In [ ]:
#icici bank
#Downloadin the stock prices for the security monthly data last 5 years
icici_df = yfin.download('ICICIBANK.NS',
                      start='2017-11-01',
                      end='2022-11-01',
                      progress=False,interval ='1mo'
)
#Calculate log returns for icici bank
icici_df['Return_ICICI'] = npy.log(icici_df['Adj Close']/icici_df['Adj Close'].shift(1))
icici_df1 = icici_df.dropna(axis=0)
icici_df1.head()
print(icici_df['Return_ICICI'])
icici_df1 = icici_df.dropna(axis=0)
icici_df1_returns = icici_df1['Return_ICICI']
icici_df1_returns.head()
returns = pds.merge(NIFTY_df1_returns,icici_df1_returns,how='inner',on='Date')
returns.head()
returns.describe()
x = returns['Return_NIFTY']
y = returns['Return_ICICI']
x_sm = stm.add_constant(x)
#Regression --- Market model
model = stm.OLS(y,x_sm)
results = model.fit()
#Regression result(summary)
results.summary()


Date
2017-11-01         NaN
2017-12-01    0.020755
2018-01-01    0.116933
2018-02-01   -0.119325
2018-03-01   -0.118122
2018-04-01    0.020799
2018-05-01    0.005614
2018-06-01   -0.037068
2018-07-01    0.099625
2018-08-01    0.129832
2018-09-01   -0.109992
2018-10-01    0.150004
2018-11-01    0.000422
2018-12-01    0.013980
2019-01-01    0.011869
2019-02-01   -0.040028
2019-03-01    0.134352
2019-04-01    0.017327
2019-05-01    0.038985
2019-06-01    0.031136
2019-07-01   -0.029014
2019-08-01   -0.030971
2019-09-01    0.057050
2019-10-01    0.065482
2019-11-01    0.101661
2019-12-01    0.050034
2020-01-01   -0.024894
2020-02-01   -0.055543
2020-03-01   -0.429121
2020-04-01    0.160594
2020-05-01   -0.135582
2020-06-01    0.057083
2020-07-01   -0.013319
2020-08-01    0.129124
2020-09-01   -0.106459
2020-10-01    0.101378
2020-11-01    0.187044
2020-12-01    0.122525
2021-01-01    0.003638
2021-02-01    0.107174
2021-03-01   -0.026530
2021-04-01    0.031120
2021-05-01    0.098635
2021-0

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           Return_ICICI   R-squared:                       0.639
Model:                            OLS   Adj. R-squared:                  0.632
Method:                 Least Squares   F-statistic:                     100.8
Date:                Sun, 20 Nov 2022   Prob (F-statistic):           3.24e-14
Time:                        17:31:15   Log-Likelihood:                 83.462
No. Observations:                  59   AIC:                            -162.9
Df Residuals:                      57   BIC:                            -158.8
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            0.0048      0.008      0.608      0.546      -0.011       0.021
Return_NIFTY     1.3685      0.136     10.041      0.000       1.096       1.641
==============================================================================
Omnibus:                        9.116   Durbin-Watson:                   2.454
Prob(Omnibus):                  0.010   Jarque-Bera (JB):                9.917
Skew:                           0.659   Prob(JB):                      0.00702
Kurtosis:                       4.515   Cond. No.                         17.5
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

- *Beta levered for ICICI Bank is 1.3685*
- *D/E ratio for ICICI bank is 6.89*
- *Unlevered Beta for ICICI Bank is =  1.3685/(1+.65X(Debt/MVE)) = 1.3685/(1+.65X(6.89))=0.2498*

In [ ]:
#kotak bank
#Downloading the stock prices for the security monthly data last 5 years
kotak_df = yfin.download('KOTAKBANK.NS',
                      start='2017-11-01',
                      end='2022-11-01',
                      progress=False,interval ='1mo'
)
#Calculating the log returns for kotak bank
kotak_df['Return_KOTAK'] = npy.log(kotak_df['Adj Close']/kotak_df['Adj Close'].shift(1))
kotak_df1 = kotak_df.dropna(axis=0) # removing the null values
kotak_df1.head()
print(kotak_df['Return_KOTAK'])
kotak_df1 = kotak_df.dropna(axis=0)
kotak_df1_returns = kotak_df1['Return_KOTAK']
kotak_df1_returns.head()
returns = pds.merge(NIFTY_df1_returns,kotak_df1_returns,how='inner',on='Date')
returns.head()
returns.describe()
x = returns['Return_NIFTY']
y = returns['Return_KOTAK']
x_sm = stm.add_constant(x)
#Regression --- Market model
model = stm.OLS(y,x_sm)
results = model.fit()
#Regression result(summary)
results.summary()


Date
2017-11-01         NaN
2017-12-01    0.009848
2018-01-01    0.093265
2018-02-01   -0.016731
2018-03-01   -0.039989
2018-04-01    0.144836
2018-05-01    0.097103
2018-06-01    0.006237
2018-07-01   -0.027249
2018-08-01   -0.014601
2018-09-01   -0.120077
2018-10-01   -0.019861
2018-11-01    0.097651
2018-12-01    0.018110
2019-01-01   -0.000597
2019-02-01   -0.034554
2019-03-01    0.095378
2019-04-01    0.038262
2019-05-01    0.093305
2019-06-01   -0.030043
2019-07-01    0.028334
2019-08-01   -0.059187
2019-09-01    0.138718
2019-10-01   -0.043468
2019-11-01    0.025521
2019-12-01    0.041921
2020-01-01    0.004384
2020-02-01   -0.043121
2020-03-01   -0.223321
2020-04-01    0.046103
2020-05-01   -0.103300
2020-06-01    0.105691
2020-07-01    0.003888
2020-08-01    0.025732
2020-09-01   -0.099837
2020-10-01    0.198977
2020-11-01    0.209008
2020-12-01    0.045361
2021-01-01   -0.152728
2021-02-01    0.038593
2021-03-01   -0.015481
2021-04-01   -0.002399
2021-05-01    0.033125
2021-0

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           Return_KOTAK   R-squared:                       0.464
Model:                            OLS   Adj. R-squared:                  0.455
Method:                 Least Squares   F-statistic:                     49.36
Date:                Sun, 20 Nov 2022   Prob (F-statistic):           2.89e-09
Time:                        17:37:08   Log-Likelihood:                 84.052
No. Observations:                  59   AIC:                            -164.1
Df Residuals:                      57   BIC:                            -159.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            0.0012      0.008      0.153      0.879      -0.014       0.017
Return_NIFTY     0.9480      0.135      7.025      0.000       0.678       1.218
==============================================================================
Omnibus:                        0.578   Durbin-Watson:                   2.076
Prob(Omnibus):                  0.749   Jarque-Bera (JB):                0.424
Skew:                           0.206   Prob(JB):                        0.809
Kurtosis:                       2.939   Cond. No.                         17.5
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

- *Beta levered for KOTAK Bank is .9480*
- *D/E ratio for KOTAK bank is 3.76*
- *Unlevered Beta for KOTAK Bank is =  0.9480/(1+.65X(Debt/MVE)) = 0.9480/(1+.65X3.76)=0.2753*

Unlevered Beta's of similar companies in the industry:

AXIS BANK:0.2623

HDFC BANK:0.1917

ICICI BANK:0.2498

KOTAK BANK:0.2753

Average Beta unlevered= (0.2653+0.1917+0.2498+0.2753)/4 =0.244

Relevering the beta for SBI:

D/E for SBI BANK is 14.8

Beta levered=0.244(1+0.65(14.8))=2.5913

Cost of equity from bottom up approach= 0.0475 + 2.59*(0.0757) = 0.2435 i.e 24.35%


# **Calculating the Cost of Debt:**

In [ ]:
#To calculate the interest rates(pre-tax cost of debts) according to the estimated ratings
def rating_spread(int_cov):
    #For the firms with market capitalisation > $5billion
    if int_cov > 8.5:
        #Rating is AAA
        creditrating_spread = 0.0069
    elif (int_cov > 6.5) & (int_cov <= 8.5):
        #Rating is AA
        creditrating_spread = 0.0085
    elif (int_cov > 5.5) & (int_cov <= 6.5):
        #Rating is A+
        creditrating_spread = 0.0107
    elif (int_cov > 4.25) & (int_cov <= 5.49):
        #Rating is A
        creditrating_spread = 0.0118
    elif (int_cov > 3) & (int_cov <= 4.25):
        #Rating is A-
        creditrating_spread = 0.0133
    elif (int_cov > 2.5) & (int_cov <= 3):
        #Rating is BBB
        creditrating_spread = 0.0171
    elif (int_cov > 2.25) & (int_cov <= 2.5):
        #Rating is BB+
        creditrating_spread = 0.0231
    elif (int_cov > 2) & (int_cov <= 2.25):
        #Rating is BB
        creditrating_spread = 0.0277
    elif (int_cov > 1.75) & (int_cov <= 2):
        #Rating is B+
        creditrating_spread = 0.0405
    elif (int_cov > 1.5) & (int_cov <= 1.75):
        #Rating is B
        creditrating_spread = 0.0486
    elif (int_cov > 1.25) & (int_cov <= 1.5):
        #Rating is B-
        creditrating_spread = 0.0594
    elif (int_cov > 0.8) & (int_cov <= 1.25):
        #Rating is CCC
        creditrating_spread = 0.0946
    elif (int_cov > 0.65) & (int_cov <= 0.8):
        #Rating is CC
        creditrating_spread = 0.0997
    elif (int_cov > 0.2) & (int_cov <= 0.65):
        #Rating is C
        creditrating_spread = 0.1309
    elif int_cov <= 0.2:
        #Rating is D
        creditrating_spread = 0.1744
    return creditrating_spread

In [ ]:
!pip install numpy_financial
import numpy_financial as npfin
#inputs related to the data of the firm and other variables
#All the amounts are in crores

rf = 0.0475                                    #Risk free rate
rp = 0.0757                                         #Market Risk Premium (Rm - Rf)
tax_rate = 0.35
marketcap = 537886.35                                    #No. of outstanding shares*share price
#T_debt = 4536570.38                                         #Number of years for interest payments on debt
debt_outstanding = 4536570.38
interest_expense = 166470.46
EBIT = 224085.50
interest_coverage_ratio = EBIT/interest_expense
spread = rating_spread(interest_coverage_ratio)
cod = rf + spread                                         #Cost of debt
spread
cod
rating_spread(1.49)

#firm_value = marketcap + mdebt
#beta_unlevered = beta_equity/(1+DE*(1-tax_rate))
#coe = rf + beta_equity*rp                       #Cost of equity
#wacc = cod*(1-tax_rate)*DV + coe*EV
#print(wacc)

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


NameError: ignored

- Interest coverage ratio= 1.35

- Credit spread=0.0594

- Cost of Debt=Rf+credit spread=0.0475+0.0594=0.1069 =10.69%


- Cost of equity from top down approach= 15.49%
- Cost of equity from bottom up aprroach= 24.35%
- D/E of SBI Bank =14.8

- E/S of SBI Bank=0.0633

- D/S of SBI Bank = 0.9367

## - WACC using the top down approach:
- *WACC= COD(1-tax rate)(D/S) + COE(E/S) =.1069(1-0.65)(0.9367)+.1549(0.0633) = 0.0749 = 7.49%*

## - WACC using the bottom up approach:
- *WACC= COD(1-tax rate)(D/S) + COE(E/S)
=.1069(1-0.65)(0.9367)+.2435(0.0633) = 0.0805 = 8.05%*






